crates/sequencer/src/sequencer.rs

```diff
@@ -866,16 +866,15 @@ where
         // and only resume normal operations once the sequencer has caught up.
         let mut missed_da_blocks_count = 0;
 
-        loop {
-            let block_production_tick = tokio::time::sleep(
-                target_block_time
-                    .checked_sub(parent_block_exec_time)
-                    .unwrap_or_default(),
-            );
+        let mut block_production_tick = tokio::time::interval(
+            target_block_time
+                .checked_sub(parent_block_exec_time)
+                .unwrap_or_default(),
```

Comment:
This is not going to work. Either keep the tick inside the loop to account for the parent block's execution time or set a constant value here.

Reply:
Why?

Reply:
Because outside the loop, `target_block_time` is always going to be 2 seconds. In this case, we should remove `checked_sub` because it doesn't make sense.
Inside the loop, `checked_sub` is going to be useful since `parent_block_exec_time` is going to contain a value larger than 0.

Reply:
Ah yeah, outside the loop we don’t need checked sub + parent exec time is 0

Reply:
I mean at that point